### PR TITLE
Optimized fneg and dneg evaluator on X86

### PR DESCRIPTION
**Pull request by 0dvictor**
Reduced one instruction (out of two instructions)
in generated code for fneg and dneg.

Signed-off-by: Victor Ding <dvictor@ca.ibm.com>

### CODE REVIEW

File: compiler/x/codegen/FPTreeEvaluator.cpp [hunk 1]
```diff
@@ -591,6 +591,63 @@ TR::Register *OMR::X86::TreeEvaluator::fpBinaryArithmeticEvaluator(TR::Node
       }
    }
 
+TR::Register *OMR::X86::TreeEvaluator::fpNegEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   static uint8_t MASKOFFLOAT[] =
```
Comment (0xdaryl) [inline compiler/x/codegen/FPTreeEvaluator.cpp]: Ideally, these arrays would be forced 16-byte aligned to improve the instruction that consumes them as a 16-byte quantity.  The C++11 `alignas` specifier would do the trick, but sadly not supported on all our build compilers. :-(

Comment (0xdaryl) [reply compiler/x/codegen/FPTreeEvaluator.cpp]: Oops.  Wrong PR, but it applies to this one too.

Comment (0dvictor) [reply compiler/x/codegen/FPTreeEvaluator.cpp]: Agreed. Once all our build compilers have the proper support, we should update these arrays with `alignas`.

**Comment by 0dvictor** [comment]
Yes, they are nearly identical. Let me mark this as WIP and common the two logics once PR #2415 is merged.

**Comment by 0dvictor** [comment]
I have updated this PR to adopt the common logic introduced by PR #2415.

**Comment by 0xdaryl** [comment]
This is an x86-only change, so a genie build is not required.

**Pull request merged**